src/bootstrap/flags.rs

```diff
@@ -256,14 +256,28 @@ To learn more about a subcommand, run `./x.py <command> -h`
             }
         });
 
+        let build = m.opt_str("build").unwrap_or_else(|| {
+            env::var("BUILD").unwrap()
+        });
+
+        // Match the `config.toml`-driven configuration behavior of
+        // implicitly inserting the build triple into host list.
+        // This is needed to make `x.py dist --host=some-host` invocations
+        // *without the build triple* to continue to work.
+        let host = {
+            let mut host = m.opt_strs("host");
+            if !host.contains(&build) {
+                host.push(build.clone());
+            }
+            host
+        };
```

Comment:
I think I'd prefer to omit this change to keep the invocations of the build system predictable by ensuring that `--host` always overrides the list of hosts. I also think it's fine if `./x.py dist --host ...` doesn't work as precisely what dist packages are created when is already kinda squishy, and I'm fine just saying that the invocation for doing that is slightly different (e.g. doesn't require this workaround)